File: server/models/student.model.js
const mongoose = require("mongoose");

const StudentSchema = mongoose.Schema(
  {
    firstName: {
      type: String,
      required: true,
    },

    lastName: {
      type: String,
      required: true,
    },

    course: {
      type: String,
      required: true,
    },

    year: {
      type: String,
      required: true,
    },

    enrolled: {
      type: Boolean,
      default: false,
    },
  },
  {
    timestamps: true,
  }
);

const Student = mongoose.model("student", StudentSchema);

module.exports = Student;

File: group_activity/lib/features/student/mappers/student_model.dart
class StudentModel {
  final String firstName;
  final String lastName;
  final String year;
  final String course;
  final bool enrolled;
  final String id;

  StudentModel({
    required this.id,
    required this.firstName,
    required this.lastName,
    required this.year,
    required this.course,
    required this.enrolled,
  });

  factory StudentModel.fromJson(Map<String, dynamic> json) {
    return StudentModel(
      id: json["_id"],
      firstName: json["firstName"],
      lastName: json["lastName"],
      year: json["year"],
      course: json["course"],
      enrolled: json["enrolled"],
    );
  }
}
